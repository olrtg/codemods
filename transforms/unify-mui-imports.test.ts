import { defineSnapshotTest } from 'jscodeshift/src/testUtils'
import transformer from './unify-mui-imports'

describe('unify mui imports', () => {
  defineSnapshotTest(
    transformer,
    {},
    `import Box from '@mui/material/Box'
    import Button from '@mui/material/Button'
    import Grid from '@mui/material/Grid'`,
    'converts to named imports',
  )

  defineSnapshotTest(
    transformer,
    {},
    `import Box from '@mui/material/Box'
    import Button from '@mui/material/Button'
    import { Grid } from '@mui/material'`,
    'converts to named imports using the existent import',
  )

  defineSnapshotTest(
    transformer,
    {},
    `import Box from '@mui/material/Box'
    import AliasedButton from '@mui/material/Button'
    import Grid from '@mui/material'`,
    'converts to named imports while using keeping the alias',
  )

  defineSnapshotTest(
    transformer,
    {},
    `import Box from '@mui/material/Box'
    import AliasedButton from '@mui/material/Button'
    import { Grid } from '@mui/material'`,
    'converts to named imports using the existent import while keeping the alias',
  )
})
