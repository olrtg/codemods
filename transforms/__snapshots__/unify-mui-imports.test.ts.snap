// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unify mui imports converts to named imports 1`] = `"import { Box, Button, Grid } from "@mui/material";"`;

exports[`unify mui imports converts to named imports using the existent import 1`] = `"import { Grid, Box, Button } from '@mui/material';"`;

exports[`unify mui imports converts to named imports using the existent import while keeping the alias 1`] = `"import { Grid, Box, Button as AliasedButton } from '@mui/material';"`;

exports[`unify mui imports converts to named imports while using keeping the alias 1`] = `"import Grid, { Box, Button as AliasedButton } from '@mui/material';"`;

exports[`unify mui imports transforms correctly 1`] = `"import { style } from "@mui/material";"`;
